src/web/src/app.tsx

```diff
@@ -0,0 +1,362 @@
+    </div>
+  );
+}
+
+export function CurrentStudyList(props: {
+  studies: Map<SeedType, StudyListModel>;
+}): JSX.Element {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const setParam = (param: string, value: string): void => {
+    setSearchParams((prev) => {
+      prev.set(param, value);
+      return prev;
+    });
+  };
+  const currentSeed = getCurrentSeedType(searchParams);
+
+  const filter = new StudyFilter();
+  filter.search = searchParams.get('search') ?? undefined;
+  filter.minPriority =
+    currentSeed === SeedType.UPSTREAM
+      ? StudyPriority.STABLE_MIN
+      : StudyPriority.NON_INTERESTING;
+  try {
+    const priorityString = searchParams.get('minPriority');
+    if (priorityString != null) {
+      filter.minPriority = parseInt(priorityString);
+    }
+  } catch {
+    /* empty */
+  }
+  filter.showEmptyGroups = searchParams.get('showEmptyGroups') === 'true';
+  const toggleShowEmptyGroups = (): void => {
+    setParam('showEmptyGroups', filter.showEmptyGroups ? 'false' : 'true');
+  };
+
+  filter.includeOutdated = searchParams.get('includeOutdated') === 'true';
+  const toggleIncludeOutdated = (): void => {
+    setParam('includeOutdated', filter.includeOutdated ? 'false' : 'true');
+  };
+
+  const studyList = props.studies
+    .get(currentSeed)
+    ?.studies(filter)
+    .map((study, i) => (
+      <StudyItem key={study.name() + i} study={study} filter={filter} />
+    ));
+
+  return (
+    <div className="row">
+      <div className="col-sm-12">
+        <div className="card mb-3">
+          <div className="card-header filter-list">
+            <FilterCheckbox
+              title="Show empty groups"
+              htmlName="showEmptyGroups"
+              checked={filter.showEmptyGroups}
+              toggle={toggleShowEmptyGroups}
+            />
+            <FilterCheckbox
+              title="Show outdated studies"
+              htmlName="includeOutdated"
+              checked={filter.includeOutdated}
+              toggle={toggleIncludeOutdated}
+            />
+          </div>
+        </div>
+        {studyList}
+      </div>
+    </div>
+  );
+}
+
+class AppState {
+  studies = new Map<SeedType, StudyListModel>();
+}
+
+export function App(): JSX.Element {
+  const [state, setState] = React.useState(new AppState());
+  React.useEffect(() => {
+    const load = async (url: string, type: SeedType): Promise<void> => {
+      const studyList = await loadSeed(url, type);
+      setState((prevState) => {
+        const newState: AppState = { ...prevState };
+        if (studyList !== undefined) newState.studies.set(type, studyList);
+        return newState;
+      });
+    };
+    load(core_utils.variationsProductionUrl, SeedType.PRODUCTION).catch(
+      console.error,
+    );
+    load(core_utils.variationsStagingUrl, SeedType.STAGING).catch(
+      console.error,
+    );
+    load(core_utils.variationsUpstreamUrl, SeedType.UPSTREAM).catch((e) => {});
+  }, []);
```

Comment:
This file is getting long and it's partly due to the support functions for this part. How about we separate UI from functional. You can put keep all the React UI functions here (although you might want to split at least some of them to new files), and then you can create a file (or directory) called `hooks` where you put all your `useXYZ` calls, and these functions that are used by them. To start, that file could export something like this that your React `App` function would use:

```tsx
// ...helper functions (loadSeed, etc)...
 
export default function useAppState() {
  const [state, setState] = React.useState(new AppState());
  React.useEffect(() => {
    const load = async (url: string, type: SeedType): Promise<void> => {
      const studyList = await loadSeed(url, type);
      setState((prevState) => {
        const newState: AppState = { ...prevState };
        if (studyList !== undefined) newState.studies.set(type, studyList);
        return newState;
      });
    };
    load(core_utils.variationsProductionUrl, SeedType.PRODUCTION).catch(
      console.error,
    );
    load(core_utils.variationsStagingUrl, SeedType.STAGING).catch(
      console.error,
    );
    load(core_utils.variationsUpstreamUrl, SeedType.UPSTREAM).catch((e) => {});
  }, []);
  return {
    state
  }
}
```

Should be a simple copy